### PR TITLE
Rewrite countPoolCandidates api query with custom methods

### LINKED ISSUE #1002: API - Pool candidates search doesn't filter multiple CmoAssets correctly
# Description
The countPoolCandidates query makes use of the whereConditions and whereHasConditions directives. However, whereHasConditions cannot quite produce the query we actually want.

For cmoAssets, we want to search for candidates who have *all* the specified skills. This is impossible with whereHasConditions. We can search for uses who have *any* of the specified skills by using the IN operator or an OR array.

`hasCmoAssets: { column: KEY, operator: IN, value: ["app_dev", "enterprise_architecture"] }` 
`hasCmoAssets: { OR: [ {column: KEY, value: "app_dev"}, {column: KEY, value: "enterprise_architecture"} ] }`)

But if we try to use an AND array to require multiple skills, the query actually tries to search for candidates who have a single CmoAsset which satisfies all the AND conditions, which is impossible, and always returns 0 candidates.

# Notes on solution
I believe the best way to resolve this issue is replace the directive-generated query with a hand-written method which handles the entire search. This will have the additional advantage of simplifying the schema definition of the countPoolCandidates query, and allowing us to make the input args similar to the PoolCandidateFilter type.

### PR DESCRIPTION
**Pull request by tristan-orourke**
Resolves #1002

### CODE REVIEW

File: api/graphql/schema.graphql [hunk 1]
```diff
@@ -176,6 +176,33 @@ type PoolCandidateSearchRequest {
...
+}
+
+input PoolFilterInput {
+    id: ID!
+}
+
+input PoolCandidateFilterInput {
+    classifications: [ClassificationFilterInput] @builder(method: "App\\Models\\PoolCandidate@filterByClassifications")
+    cmoAssets: [KeyFilterInput] @builder(method: "App\\Models\\PoolCandidate@filterByCmoAssets")
+    hasDiploma: Boolean @where(key:  "has_diploma")
```
Comment (mnigh) [inline api/graphql/schema.graphql:195]:
```suggestion
    hasDiploma: Boolean @where(key: "has_diploma")
```

**Review by mnigh — CHANGES_REQUESTED** [review]
Nice work! A few small suggestions to address and then ready to be approved.

**Comment by tristan-orourke** [comment]
Note to self: make sure that false values for the booleans don't exclude people from the search.

**Comment by tristan-orourke** [comment]
@mnigh I accepted your suggestions but also made a new commit that I forgot the first time

**Review by mnigh — APPROVED** [review]
👍

**Pull request merged**